engine/table/src/main/java/io/deephaven/engine/table/impl/asofjoin/BucketedChunkedAjMergedListener.java

```diff
@@ -791,7 +793,17 @@ private RowSet indexFromBuilder(int slotIndex) {
     @Override
     protected void destroy() {
         super.destroy();
-        leftSsaFactory.close();
-        rightSsaFactory.close();
+        if (getUpdateGraph().clock().currentState() == LogicalClock.State.Idle) {
```

Comment:
We're repeating this pattern, do we think it is useful enough to make a helper for it (which would also have the benefit of a Javadoc comment that explains the reasoning, etc.).